src/scripts/import-sdr-shipments-sheet/add-countries.ts

```diff
@@ -0,0 +1,260 @@
+  console.log("Adding Geo.Countries from SDR Shipments sheet...");
+
+  const uniqueCountries = [...consolidateCountries(shipments)];
+
+  const results = await Promise.allSettled<CountryUploadWorkflow>(
+    uniqueCountries.map((country) => {
+      return new Promise<CountryUploadWorkflow>((resolve, _reject) => {
+        resolve({
+          data: {},
+          orig: country,
+          status: UploadWorkflowStatus.PROCESSING,
+          logs: [],
+        });
+      })
+        .then(parseCountry)
+        .then(getCountry)
+        .then(uploadCountry);
+    }),
+  );
+
+  // { "SUCCESS": [], "ALREADY_EXISTS": [], ...}
+  const resultsMap: CountryUploadWorkflowResults = Object.keys(
+    UploadWorkflowStatus,
+  ).reduce((resultsMap, key) => {
+    resultsMap[key] = [];
+    return resultsMap;
+  }, {} as CountryUploadWorkflowResults);
+
+  results
+    .map((result) => {
+      if (isFulfilled(result)) {
+        return result.value;
+      } else {
+        return result.reason;
+      }
+    })
+    .reduce((resultsMap, workflowResult) => {
+      if (workflowResult.status) {
+        resultsMap[workflowResult.status].push(workflowResult);
+      } else {
+        resultsMap[UploadWorkflowStatus.OTHER].push(workflowResult);
+      }
+      return resultsMap;
+    }, resultsMap);
+
+  console.log("Add Geo.Countries results:");
+  Object.keys(resultsMap).forEach((key) => {
+    console.log(`    ${key}: ${resultsMap[key].length}`);
+
+    // NOTE: uncomment & set the status key to debug different types of results
+    // if (key !== UploadWorkflowStatus.SUCCESS && key !== UploadWorkflowStatus.ALREADY_EXISTS) {
+    //   resultsMap[key].forEach((result) => {
+    //     console.log(result)
+    //     console.log("\n")
+    //   })
+    // }
+  });
+
+  console.log("Adding items completed!");
+
+  const validCountries: Country[] = [
+    ...resultsMap[UploadWorkflowStatus.SUCCESS],
+    ...resultsMap[UploadWorkflowStatus.ALREADY_EXISTS],
+  ].reduce((countries: Country[], workflow: CountryUploadWorkflow) => {
+    return [...countries, workflow.data];
+  }, [] as Country[]);
+
+  return validCountries;
+}
+
+const isFulfilled = <T>(
+  value: PromiseSettledResult<T>,
+): value is PromiseFulfilledResult<T> => {
+  return value.status === "fulfilled";
+};
+
+const _isRejected = <T>(
+  value: PromiseSettledResult<T>,
+): value is PromiseRejectedResult => {
+  return value.status === "rejected";
+};
+
+/* Consolidate Countries
+ * ------------------------------------------------------ */
+function consolidateCountries(shipments: ShipmentCsv[]) {
+  const countries = shipments.reduce((countries: Set<string>, shipment) => {
+    if (shipment.sendingCountry !== "") {
+      countries.add(shipment.sendingCountry);
+    }
+
+    if (shipment.receivingCountry !== "") {
+      countries.add(shipment.receivingCountry);
+    }
+
+    return countries;
+  }, new Set<string>());
+
+  return countries;
+}
```

Comment:
With lodash this could be:

```typescript
chain(shipments).
    map(s => s.sendingCountry || s.receivingCountry || undefined).
    filter(isDefined).
    uniq().
    value()
```

And `isDefined` is:

```typescript
export const isDefined = <T,>(v: T | undefined | null): v is T =>
  v !== undefined && v !== null;
```